students/rmart300/mailroom/mailroom.py

```diff
@@ -38,31 +41,47 @@ def create_report():
     
     #sort the dictionary by descending order of the sum of values 
     sorted_dict = sorted(donation_dict.items(), key=lambda x: sum(int(v) for v in x[1]),reverse=True)
-    for donor, donation_list in sorted_dict:
-        donation_count = len(donation_list)
-        donation_total = sum(int(v) for v in donation_list)
+    for donor, amount_list in sorted_dict:
+        donation_count = len(amount_list)
+        donation_total = sum(int(v) for v in amount_list)
         donation_average = round(donation_total/donation_count,2)
         data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, str(donation_total), str(donation_count), str(donation_average))
         print(data_row)
-        
-    
+
+def write_letters():
+   for donor in donor_dict:

```

Comment:
instead of accessing by key later it is more pythonic to iterator over keys and values:
```
for donor, donor_data in donor_dict.items():
```